File: lua/cat.lua
#!/usr/bin/env lua
while true do
    local slice = io.read(1024)
    if slice then
        if not io.write(slice) then
            io.stderr:write("Failed to write to standard output\n")
            os.exit(false)
        end
    else
        break
    end
end

File: julia/cat.jl
#!/usr/bin/env julia
count = 0
while true
    bytes = read(stdin, 1024)
    if length(bytes) == 0 break; end
    written = write(stdout, bytes);
    if written != length(bytes)
        println("Failed to write all bytes to stdout");
        exit(1)
    end
end
